### DISCUSSION TITLE
Confusion about Normalisation of Dataset.

### DISCUSSION THREAD

**Discussion by anubhavkamal** [MilesCranmer/PySR#452] [Q&A] [answered]
Hello!

PySR has turned out to be a hidden gem for us recently on certain problems. 
While testing on one of the datasets, the results (analytical solution) were fairly simple and more accurate when it ran on data without normalisation and the accuracy dipped slightly when we normalised the data. 

I might be drawing a parallel to NN here but do we need to normalise our data for Symbolic regression in general? or both of the approaches are equivalent?
Thanks

**Comment by MilesCranmer** [comment, marked as answer]
Hi @anubhavkamal,

You don’t *need* to normalize data in PySR, as it can find constants of any magnitude, including scales that are very large or small. However, if you are okay with the extra complexity of normalization constants, then it can definitely help the search if you normalize. The reason for this is that new constants in the equations are initialized with a call to `randn()` so it can be preferable for the initial constants to be similar magnitude to the data. But it’s highly problem dependent!

cheers,
Miles

**Comment by anubhavkamal** [comment]
Thank you for the answer!
It helped!
